src/overloads.jl

```diff
@@ -0,0 +1,29 @@
+# Convenience functions for PJ_COORD
+
+Base.length(::PJ_COORD) = 4
+
+Base.@propagate_inbounds function Base.getindex(coord::PJ_COORD, i::Int)
```

Comment:
Rather than doing this, you could perhaps consider making `PJ_COORD` into a `FieldVector` subtype. Then it becomes a proper `StaticArray`. Would require tweaking the clang-based code generator.

Reply:
Though having said that, I'm not 100% sure that it's desired to have all array operations defined on this. PJ_COORD isn't really a mathematical vector at all.

Reply:
Can't it be thought of as a 4-vector (or, at least, a 4-point)?

Reply:
It's a space time 4-point, yes, but generally has no vector properties or even affine space properties (so we're back to the usual LinearAlgebra-object vs array abstraction problem). It's certainly an array of components though so making it an `AbstractArray` would be no more terrible than other non-vector things which get stored in `Vector` :-)